internal/distributed/utils/util.go

```diff
@@ -30,3 +35,67 @@ func GracefulStopGRPCServer(s *grpc.Server) {
 		<-ch
 	}
 }
+
+func getTLSCreds(certFile string, keyFile string, nodeType string) credentials.TransportCredentials {
+	log.Info("TLS Server PEM Path", zap.String("path", certFile))
+	log.Info("TLS Server Key Path", zap.String("path", keyFile))
+	creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+	if err != nil {
+		log.Warn(nodeType+" can't create creds", zap.Error(err))
+		log.Warn(nodeType+" can't create creds", zap.Error(err))
+	}
+	return creds
+}
+
+func EnableInternalTLS(NodeType string) grpc.ServerOption {
+	var Params *paramtable.ComponentParam = paramtable.Get()
+	var serverCfg *paramtable.GrpcServerConfig
+	switch strings.ToLower(NodeType) {
+	case "datacoord":
+		serverCfg = &Params.DataCoordGrpcServerCfg
+	case "datanode":
+		serverCfg = &Params.DataNodeGrpcServerCfg
+	case "indexnode":
+		serverCfg = &Params.IndexNodeGrpcServerCfg
+	case "proxy":
+		serverCfg = &Params.ProxyGrpcServerCfg
+	case "querycoord":
+		serverCfg = &Params.QueryCoordGrpcServerCfg
+	case "querynode":
+		serverCfg = &Params.QueryNodeGrpcServerCfg
+	case "rootcoord":
+		serverCfg = &Params.RootCoordGrpcServerCfg
```

Comment:
The code here is a bit awkward. The `startGrpcLoop` function for each of the components calls _this_ function by passing in a string indicating the component name, and then this function uses a switch to decide which component param to use.

However, in `milvus.yaml`, you have one common section for internalTLS configs. To match with this I suggest that you do one of the following:
- Remove the common internal TLS parameters from [`grpcConfig`](https://github.com/milvus-io/milvus/blob/2b3f5bec070a49895e801df56567b7ca9bf60741/pkg/util/paramtable/grpc_param.go#L66) and move them to [`commonCfg`](https://github.com/milvus-io/milvus/blob/2b3f5bec070a49895e801df56567b7ca9bf60741/pkg/util/paramtable/component_param.go#L191), or
- Create a new config struct (say, `internalTLSConfig`) and add it to [`ComponentParam`](https://github.com/milvus-io/milvus/blob/2b3f5bec070a49895e801df56567b7ca9bf60741/pkg/util/paramtable/component_param.go#L61)

One possible drawback of this approach is that you won't be able to specify different certificate paths for the different nodes, but this shouldn't be a problem. If the nodes are deployed in separate containers / machines, the same path can have different certificates, and if the nodes are on the same machine (for example in standalone) a single certificate should be enough.

Reply:
I can create a struct and do  what you suggest, but the TLS configs are also part of the grpc configs. Hence, I added internalTLS configs here.Let me add a new struct to make this better.

Reply:
> The code here is a bit awkward. The `startGrpcLoop` function for each of the components calls _this_ function by passing in a string indicating the component name, and then this function uses a switch to decide which component param to use.
> 
> However, in `milvus.yaml`, you have one common section for internalTLS configs. To match with this I suggest that you do one of the following:
> 
> * Remove the common internal TLS parameters from [`grpcConfig`](https://github.com/milvus-io/milvus/blob/2b3f5bec070a49895e801df56567b7ca9bf60741/pkg/util/paramtable/grpc_param.go#L66) and move them to [`commonCfg`](https://github.com/milvus-io/milvus/blob/2b3f5bec070a49895e801df56567b7ca9bf60741/pkg/util/paramtable/component_param.go#L191), or
> * Create a new config struct (say, `internalTLSConfig`) and add it to [`ComponentParam`](https://github.com/milvus-io/milvus/blob/2b3f5bec070a49895e801df56567b7ca9bf60741/pkg/util/paramtable/component_param.go#L61)
> 
> One possible drawback of this approach is that you won't be able to specify different certificate paths for the different nodes, but this shouldn't be a problem. If the nodes are deployed in separate containers / machines, the same path can have different certificates, and if the nodes are on the same machine (for example in standalone) a single certificate should be enough.

Thanks for the comment, I'm just gonna to point out the same thing